src/components/FormSections/LocationDetails.tsx

```diff
@@ -159,8 +172,12 @@ const LocationDetails: React.FC = () => {
 						name="streetAddress"
 						size="full"
 						disabled={isDisabled}
-						onChange={filterOnTextChange}
+						onChange={changes => {
+							filterOnTextChange(changes)
+						}}
+						onBlur={() => trigger("city")}
 						maxLength={200}
+						ref={register}
```

Comment:
See my previous comment about actually invoking the register function
```suggestion
						ref={register()}
```